docs/contributing/component-js-configuration-options.md

```diff
@@ -0,0 +1,192 @@
+
+Configuration options straddle the line between Nunjucks, HTML and JavaScript.
+
+## Preparing the component's JavaScript
+
+First, make sure the entry function for the component has a parameter for passing in a configuration object.
+
+```javascript
+function Accordion ($module, config) {
+  ...
+}
+```
+
+Within the entry function, you'll then create an object with the default configuration for the component. These are the options the component will use if no other options are provided.
+
+```javascript
+var defaultConfig = {
+  rememberExpanded: true
+}
+```
+
+Next, use the `mergeConfigs` helper to combine the default config and the configuration object. The result of the merge is assigned globally (using `this`) so that it's accessible anywhere within the component's JavaScript.
+
+The order in which variables are written defines their priority, with objects passed sooner being overwritten by those passed later. As we want the user's configuration to take precedence over our defaults, we list our default configuration object first.
+
+There is no guarantee `config` will have any value at all, so it'll be `undefined`. We use an OR operator (`||`) as a safety check. If the value is `undefined`, we use an empty object instead.
+
+```javascript
+import { mergeConfigs } from '../../common/index.mjs'
+
+this.config = mergeConfigs(
+  defaultConfig,
+  config || {}
+)
+```
+
+We can now reference the configuration option anywhere we need to in the component's JavaScript:
+
+```javascript
+this.config.rememberExpanded
+// => true
+```
+
+It's now possible to individually initialise the component with configuration options.
+
+```html
+<script>
+  var $accordion = document.getElementById("accordion")
+  new window.GOVUKFrontend.Accordion($accordion, {
+    rememberExpanded: false
+  }).init()
+</script>
+```
+
+## Allowing options to be passed through the `initAll` function
+
+Usually, teams will not be individually initialising components. Instead, GOV.UK Frontend ships with an `initAll` function, which searches the page for instances of components and automatically initialises them.
+
+In `src/govuk/all.mjs`, update the component's `new` function to pass through a nested configuration object. The nested object should use the component's name converted to camelCase (for example, the 'Character Count' component becomes `characterCount`).
+
+```javascript
+new Accordion($accordion, config.accordion).init()
+```
+
+It's now possible to pass configuration options for your component, as well as multiple other components, using the `initAll` function.
+
+```html
+<script>
+  window.GOVUKFrontend.initAll({
+    accordion: {
+      rememberExpanded: false
+    },
+    characterCount: {
+      threshold: 90
+    }
+  })
+</script>
+```
+
+## Adding support for HTML `data-*` attributes
```

Comment:
Mind if we link to this section from the JS coding standards?
https://github.com/alphagov/govuk-frontend/blob/main/docs/contributing/coding-standards/js.md#use-data-attributes-to-initialise-component-javascript

It's not been updated beyond `data-module`